search-sentiments-map/test/search-test.js

```diff
@@ -1,35 +1,220 @@
+        async function() {
+          let mockCountryData = [
+            {
+              country: 'US',
+            },
+            {
+              country: 'GB',
+            },
+            {
+              country: 'KZ',
+            }
+          ];
+          let results =
+              await search.retrieveUserSearchResultFromDatastore(
+                  'testTopic', ['GB', 'KZ', 'US']);
+
+          assert.equal(results.topic, 'testTopic');
+          assert.equal(results.timestamp, mockData[0].timestamp);
+          assert.deepEqual(results.dataByCountry, mockCountryData);
+        });
+
+    it('should fetch and retrieve search data if no data exists',
+        async function() {
+          let mockCountryData = [
+            {
+              country: 'AU',
+            },
+            {
+              country: 'US',
+            }
+          ];
+          let results =
+              await search.retrieveUserSearchResultFromDatastore(
+                  'newTestTopic', ['AU', 'US']);
+
+          assert.equal(results.topic, 'newTestTopic');
+          assert.isAbove(results.timestamp, Date.now() - 60000);
+          assert.deepEqual(results.dataByCountry, mockCountryData);
+        });
+  });
+
   describe('UpdateSearchResults', function() {
     let deleteAncientResultsStub;
-    let retrieveGlobalTrendsStub;
-    let updateSearchResultsForTopicStub;
+    let getSearchResultsForCountriesForTopicStub;
+    let sleepStub;
+    let datastoreEntities;
+    let globalTrends;
 
     beforeEach(() => {
+      datastoreEntities = [];
+      globalTrends = [
+        {trendTopic: "trend1"},
+        {trendTopic: "trend2"},
+        {trendTopic: "trend3"},
+      ];
+
+      // Function is stubbed because it is tested separately.
       deleteAncientResultsStub = 
           sinon.stub(search, 'deleteAncientResults')
               .resolves('Not interested in the output');
-      sleepStub = 
-          sinon.stub(search, 'sleep')
-              .resolves('Not interested in the output');
-      retrieveGlobalTrendsStub = 
-          sinon.stub(search, 'retrieveGlobalTrends')
-              .resolves([
-                {trendTopic: 'trend1'},
-                {trendTopic: 'trend2'},
-              ]);
+
+
+      // Function is stubbed because it is tested separately.
       getSearchResultsForCountriesForTopicStub =
           sinon.stub(search, 'getSearchResultsForCountriesForTopic')
               .resolves('Not interested in the output');
-      addWorldDataByTopicToDatastoreStub =
-          sinon.stub(search, 'addWorldDataByTopicToDatastore')
+
+      // Sleep function is stubbed to avoid 1 minute pauses.
+      sleepStub = 
+          sinon.stub(search, 'sleep')
               .resolves('Not interested in the output');
+
+      // Calls to the datastore are stubbed.
+      sinon.stub(Datastore.prototype, 'runQuery').callsFake(() => {
+        trendsEntries = [{
+          globalTrends: globalTrends
```

Comment:
This statement doesn't look the best.

Rename the variable to globalTrendsList.